compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -324,8 +324,8 @@ void addPadPackBasedPassPipeline(OpPassManager &pm,
                                  TilingConfig &tilingConfig) {
   auto &modulePassManager = pm.nest<ModuleOp>();
   modulePassManager.addNestedPass<func::FuncOp>(createAMDAIECleanupPass());
-  pm.addPass(createCanonicalizerPass());
-  pm.addPass(createCSEPass());
+  modulePassManager.addNestedPass<func::FuncOp>(createCanonicalizerPass());
+  modulePassManager.addNestedPass<func::FuncOp>(createCSEPass());
```

Comment:
This is for my understanding - why would we want to invoke some `--canonicalize --cse` like a `nestedPass` while the others can be as it is?

Reply:
Actually do we really need to run cleanup, canonicalization, cse at the beginning? I don't think these are useful, we'll need to check and clean up these passes in the following patch.

Reply:
So about the nesting - without the nesting these cleanup passes were not called along with the nested passes and were all called in the end. So yes techincally all of them should be nested, I will update that. 
Regarding why we need it - The tiling leaves behind some dead ops from the previous state. So I ended up accidently applying padding on the untiled dead matmul rather than the new tiled matmul. So its easier to have a claen state before starting a pass.

Reply:
Just looking closely now, I think adding the nested pass is not necessary but we do have to apply the passes to modulePassManager and not just the pass manager. Also, looking at the IR yes some of these clenup passes are unnecessary so I will remove them.

Reply:
Nice, thanks for cleaning up the passes!